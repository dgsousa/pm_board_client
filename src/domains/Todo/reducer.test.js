import {
  addTodo,
  updateTodo,
  deleteTodo,
  initialState,
} from './reducer';


test('addTodo', () => {
  const action = {
    key: '123',
    val: 'Run the test',
  };
  const newState = addTodo(initialState, action);
  expect(newState.toJS()).toMatchSnapshot();
});

test('updateTodo', () => {
  const currentState = initialState.set('currentTodo', 'updated');
  const action = {
    todo: 'updated Todo',
  };

  const newState = updateTodo(currentState, action);
  expect(newState.toJS()).toMatchSnapshot();
});

test('deleteTodo', () => {
  const currentState = initialState.setIn(['todos', 'testKey'], 'testVal');
  const action = {
    key: 'testKey',
  };
  const newState = deleteTodo(currentState, action);
  expect(newState.toJS()).toMatchSnapshot();
});
