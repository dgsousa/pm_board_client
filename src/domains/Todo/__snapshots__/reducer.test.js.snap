// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addTodo 1`] = `
Object {
  "currentTodo": "",
  "todos": Object {
    "123": "Run the test",
  },
}
`;

exports[`deleteTodo 1`] = `
Object {
  "currentTodo": "",
  "todos": Object {},
}
`;

exports[`updateTodo 1`] = `
Object {
  "currentTodo": "updated Todo",
  "todos": Object {},
}
`;
